### PR TITLE
DOC: add more examples to StringMethods on Index

### PR DESCRIPTION
**Pull request by mortada**
as discussed in https://github.com/pydata/pandas/pull/9667

@jorisvandenbossche please take a look, thanks

### CODE REVIEW

File: doc/source/text.rst [hunk 1]
```diff
@@ -37,6 +37,32 @@ the equivalent (scalar) built-in string methods:
...
+transforming DataFrame column names. For instance, you may have column names
+with leading or trailing whitespaces
+
+.. ipython:: python
+
+   df = DataFrame(randn(3, 2), columns=[' Column A ', ' Column B '],
+                  index=range(3))
+   df
+
+Since the column of a DataFrame is an Index object, we can use the .str accessor
```
Comment (shoyer) [inline doc/source/text.rst]: maybe "column names" instead of "columns"?

**Comment by mortada** [comment]
@shoyer good idea, it's updated

File: doc/source/text.rst [hunk 2]
```diff
@@ -37,6 +37,32 @@ the equivalent (scalar) built-in string methods:
...
+transforming DataFrame columns. For instance, you may have columns with
+leading or trailing whitespaces
+
+.. ipython:: python
+
+   df = DataFrame(randn(3, 2), columns=[' Column A ', ' Column B '],
+                  index=range(3))
+   df
+
+Since ``df.columns`` is an Index object, we can use the .str accessor
```
Comment (shoyer) [inline doc/source/text.rst]: quote .str with two backticks?

File: doc/source/text.rst [hunk 3]
```diff
@@ -37,6 +37,32 @@ the equivalent (scalar) built-in string methods:
...
+
+.. ipython:: python
+
+   df.columns.str.strip()
+   df.columns.str.lower()
+
+These string methods can then be used to clean up the columns as neeeded.
```
Comment (shoyer) [inline doc/source/text.rst]: spelling: neeeded -> needed

File: doc/source/text.rst [hunk 4]
```diff
@@ -37,6 +37,32 @@ the equivalent (scalar) built-in string methods:
    idx.str.lstrip()
    idx.str.rstrip()
 
+The string methods on Index are especially useful for cleaning up or
+transforming DataFrame columns. For instance, you may have columns with
+leading or trailing whitespaces
```
Comment (shoyer) [inline doc/source/text.rst]: end with a colon `:`

File: doc/source/text.rst [hunk 5]
```diff
@@ -37,6 +37,32 @@ the equivalent (scalar) built-in string methods:
...
+Since ``df.columns`` is an Index object, we can use the .str accessor
+
+.. ipython:: python
+
+   df.columns.str.strip()
+   df.columns.str.lower()
+
+These string methods can then be used to clean up the columns as neeeded.
+Here we are removing leading and trailing whitespaces, lowercasing all names,
+and replacing any remaining whitespaces with underscores.
```
Comment (shoyer) [inline doc/source/text.rst]: also should end with a colon, not a period.

Comment (mortada) [inline doc/source/text.rst]: thanks! all places updated

File: doc/source/text.rst [hunk 6]
```diff
@@ -37,6 +37,32 @@ the equivalent (scalar) built-in string methods:
    idx.str.lstrip()
    idx.str.rstrip()
 
+The string methods on Index are especially useful for cleaning up or
+transforming DataFrame columns. For instance, you may have columns with
+leading or trailing whitespaces:
```
Comment (shoyer) [inline doc/source/text.rst]: Just "whitespace", I think

**Comment by shoyer** [comment]
This looks great to me! I'll just hold off on merging until Joris can take a look.

**Comment by jorisvandenbossche** [comment]
Looks good, thanks a lot!

**Pull request merged**